### PR TITLE
CLC-6074: Selenium tests for editable profile features in GL3

### PR DESCRIPTION
**Pull request by pfarestveit**
https://jira.ets.berkeley.edu/jira/browse/CLC-6074

This introduces two new tests, one for the basic info UI and another for the contact info UI.  Given the unreliability of test data, the tests for now will be run locally during regression testing and not as part of the nightly test runs.

Both tests require a JSON test data file to provide inputs and selections for the edit forms, an example being added to fixtures.  This is especially true of the contact info address tests.  The test data file will contain all the country variations of addresses that are currently configured in Campus Solutions.  Should those configurations change, only the test data file will need to be updated; the test itself will automatically adapt to the change with no code update required.

### CODE REVIEW

File: config/settings.yml [hunk 1]
```diff
@@ -367,6 +367,7 @@ ui_selenium:
   googleTaskTimeout: 12
   pageEventTimeout: 3
   pageLoadTimeout: 12
+  campus_solutions_timeout: 6
```
Comment (johncrossman) [inline config/settings.yml]: Any reason for the mix of camelCase and underscore_case?

File: fixtures/ui_selenium/profile.json [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            "basicInfo": {
+                "uid": "999999",
+                "sid": "99999999",
+                "officialName":  {
+                    "firstName": "Oski",
+                    "lastName": "Bear"
+                },
+                "preferredName":  {
+                    "option1": "Oskirrific",
+                    "option2": "Oskitrocious",
```
Comment (johncrossman) [inline fixtures/ui_selenium/profile.json]: Can we use `Oskitrocious` from now on?

File: fixtures/ui_selenium/profile.json [hunk 3]
```diff
@@ -0,0 +1,150 @@
+{
+    "users": [
```
Comment (christianvuerings) [inline fixtures/ui_selenium/profile.json]: @pfarestveit could we use 2 spaces everywhere in this file?

File: spec/ui_selenium/my_profile_contact_info_spec.rb [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+    after(:all) do
+      WebDriverUtils.quit_browser(@driver)
+    end
+
+    describe 'phone number' do
+
+      before(:all) do
+        # Get rid of any existing phone data
+        @contact_info_card.delete_all_phones
+        @possible_phone_types = %w(Local, Mobile, Home/Permanent)
```
Comment (johncrossman) [inline spec/ui_selenium/my_profile_contact_info_spec.rb]: this array is equivalent to ["Local," , "Mobile," ...]   The commas within values is expected?

**Comment by johncrossman** [comment]
Looks good. I'll merge once questions are answered

File: config/settings.yml (hunk 1 above)
Comment (pfarestveit) [inline config/settings.yml]: If it_s ok, I'd like to save that cleanUp for the post-SIS 'clean up QA's messy test code' sprint a year from now.  OK?

File: fixtures/ui_selenium/profile.json (hunk 3 above)
Comment (pfarestveit) [inline fixtures/ui_selenium/profile.json]: Done

File: spec/ui_selenium/my_profile_contact_info_spec.rb (hunk 4 above)
Comment (pfarestveit) [inline spec/ui_selenium/my_profile_contact_info_spec.rb]: Gracias.  Fixed.

**Pull request merged**